airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/state/MemoryManagerTest.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.state
+
+import io.micronaut.context.annotation.Replaces
+import io.micronaut.context.annotation.Requires
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import jakarta.inject.Singleton
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withContext
+import kotlinx.coroutines.withTimeout
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+@MicronautTest
+class MemoryManagerTest {
+    @Singleton
+    @Replaces(MemoryManager::class)
+    @Requires(env = ["test"])
+    class MockAvailableMemoryProvider : AvailableMemoryProvider {
+        override val availableMemoryBytes: Long = 1000
+    }
+
+    @Test
+    fun testReserveBlocking() = runTest {
+        val memoryManager = MemoryManager(MockAvailableMemoryProvider())
+        val reserved = AtomicBoolean(false)
+
+        try {
+            withTimeout(5000) { memoryManager.reserveBlocking(900) }
```

Comment:
nit: can we use a lock (or a channel) instead of a timeout here?

Reply:
I'm testing that it doesn't block, so I'm not sure what else I can do.